marquez/utils.py

```diff
@@ -23,19 +22,21 @@ def __new__(cls, *args, **kwargs):
                 JobIdMapping, cls).__new__(cls, *args, **kwargs)
         return cls._instance
 
-    def set(self, key, val):
+    @staticmethod
+    def set(key, val):
         airflow.models.Variable.set(key, val)
 
-    @provide_session
-    def pop(self, key, session=None):
-        q = session.query(airflow.models.Variable).filter(
-            airflow.models.Variable.key == key)
-        if not q.first():
-            return
-        else:
-            val = q.first().val
-            q.delete(synchronize_session=False)
-            return val
+    @staticmethod
+    def pop(key, session):
+        if session:
+            q = session.query(airflow.models.Variable).filter(
+                airflow.models.Variable.key == key)
+            if not q.first():
+                return
+            else:
+                val = q.first().val
+                q.delete(synchronize_session=False)
+                return val
```

Comment:
_Minor:_ Might be helpful to log or add a metric to provide insight into how often a session is lost / doesn't exists?

Reply:
Will have it in mind once we introduce monitoring, but I want to move away from using this mechanism in the future once we allow Marquez to use airflow's jobrun ids.